### PR TITLE
[MM-9577] Update pinned posts list on channel switch

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
If the RHS is opened and showing pinned posts and we switch channels then the pinned post in the RHS are updated to show the pinned posts of the selected channel

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9577

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: actions/global_actions.jsx [hunk 1]
```diff
@@ -75,6 +79,12 @@ export function emitChannelClickEvent(channel) {
             BrowserStore.setGlobalItem(Constants.PREV_CHANNEL_KEY + teamId, chan.name);
         }
 
+        // When switching to a different channel if the pinned posts is showing
+        // Update the RHS state to reflect the pinned post of the selected channel
```
Comment (lieut-data) [inline actions/global_actions.jsx]: nit: no need to capitalize `Update` in this sentence.

**Comment by enahum** [comment]
@lieut-data I thought about that but then rhs thread will also be bound to that, I’m I right?

**Comment by lieut-data** [comment]
My brain is in another part of the code at the moment, but presumably it could just ignore the channel id changing unless `showPinnedPosts`?

**Comment by enahum** [comment]
Probably, We would need to dig deeper, I’m just using what was already there, so for now I’ll say its ok, if you agree, lets merge this

**Comment by lieut-data** [comment]
Agreed :)

**Pull request merged**